### ISSUE TITLE
 Infinite increase in memory usage

### ISSUE THREAD

**Issue by strahe** [robinhood/faust#169]
## Checklist

# Versions

* Python version: 3.7
* Faust version: 1.1.1
* Operating system: alpine(docker)
* Kafka version: 0.10
* RocksDB version (if applicable)

Hi, I deployed my app using K8s, when I set the number of containers to 1, everything is fine.
However, when I set the number of containers to 2, one of them works fine, another instance will take up 100% of the single core CPU, and the memory grows indefinitely. It is eventually killed and then restarted by K8s, No error log printed.

This is the startup log for both instances and looks the same.
![s](https://user-images.githubusercontent.com/7932644/45914503-87e3b400-be78-11e8-8019-236fb163bb29.png)

This is one of the instance resource occupations(Approximately 350M of memory.).
![](//static.eosfans.tc.ink/photo/2018/bc89cb22-f44e-45b5-aa49-58b3a7381155.png!large)

This is another abnormal example, memory is constantly increasing.
![](//static.eosfans.tc.ink/photo/2018/9f2c4bb7-d506-44a6-b3ae-91e193138c09.png!large)

Thank you for providing such a good tool. Any tips given to me will be very helpful.

**Comment by vineetgoel** [comment]
Hi this is interesting. Does your application use windowed tables? Is there any issues the application is facing with respect to kafka connection.

**Comment by strahe** [comment]
@vineetgoel 
No, I didn't use the function associated with table, and, i used the kafka cloud service, the performance is enough,  I didn't find any issuses.

**Comment by vineetgoel** [comment]
@ask any idea here? 

@mwolting could you provide your application config?

**Comment by mwolting** [comment]
Config: 
```
+ƒaµS† v1.1.3----------------------------------------------+
| id        | my-app                                       |
| transport | kafka://kafka:9092 +uvloop                   |
| store     | memory:                                      |
| web       | http://localhost:6066/                       |
| log       | -stderr- (info)                              |
| pid       | 8                                            |
| hostname  | my-app-cdd76ff65-hcn2r                       |
| platform  | CPython 3.7.0 (Linux x86_64)                 |
| drivers   | aiokafka=0.4.18 aiohttp=3.4.4                |
| datadir   | /code/my-app-data                            |
| appdir    | /code/my-app-data/v1                         |
+-----------+----------------------------------------------+
```
Only config I set on app creation is app ID, store type, and Kafka broker. I've also tried with the rocksdb store, which yields the same result.

**Comment by strahe** [comment]
@ask No, I only have one kafka partition, but I am trying to start two workers. 
Do you mean that the number of partitions must equal the number of workers?

**Comment by vineetgoel** [comment]
@mwolting could you provide us an example app that we can run locally to reproduce this? That should help us find the culprit a lot better.

**Comment by zjw0358** [comment]
have same problem ,with run a sample code 

osx mojave ,python3.7 ,install kafka with brew

**Comment by amitripshtos** [comment]
I'm having that same issue with my setup:
kafka 1.0.1 (3 nodes)
faust 1.2.2

My application reads from topics to sending the data to elasticsearch (using async-elasticsearch-py)
The application works great with 5 faust workers, however when adding more than 5 it looks like it rebalancing, but then all the workers hang at:
"Waiting for ongoing recovery to stop" log

My guess is that it is related to the amount of partitions in one of my topics that causes the issue (my topics have between 1 to 20 partitions)

**Comment by mannmann2** [comment]
Facing the same issue here as well. Consuming from a topic with 9 partitions, and 3 Faust consumers. Memory continues to leak until the limits are hit. 

Faust versions- 1.5.1 & 1.10.4

**Comment by leoscalco** [comment]
Facing the same issue. Consuming from a topic with 10 partition and 1 Faust consumer. 

Faust version -  1.10.4

I'm trying some possible configurations to get less data but any working until now and always end in the same way.

**Comment by ryou90** [comment]
Please take a look at the fork faust-streaming :)
